### PR TITLE
[exporter/awsxray] Allow indexing of special attributes that have been filtered out

### PR DESCRIPTION
**Pull request by jj22ee**
**Description:** <Describe what has changed.>

If rpc.method (or aws.operation) is in the indexed_attributes list, it will not appear as an annotation.

Issue stems from:
- the makeAws function filters out aws.operation which can carry the rpc.method when aws.operation was not set
- The filtered attributes is then filtered again for some database stuff by makeSQL function
- The filtered (again) attributes finally are processed by makeXRayAttributes function which handles annotations and metadata

[Example location of attributes being filtered](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/awsxrayexporter/internal/translator/aws.go#L135-L138)

Due to the filtering, `aws.operation` or `rcp.method` attributes cannot become annotations. The reason why this is currently the case is likely to avoid duplicate data because these special attributes are already present in a different section of trace, but not indexed.

The change in this commit is to allow these special attributes to be able to be annotations when explicitly listed in `indexedAttrs`. The filtered attributes list is now checked in `addSpecialAttributes` to see if there are any missing attributes that are explicitly listed to be index that does exists in the unfiltered attributes list. These missing attributes are added into the filtered attributes list.

Note - if `indexAllAttrs` is true AND the special attributes are NOT in `indexedAttrs`, then these attributes should not be indexed/annotated.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>
Unit tests ensures that the attributes that have been filtered out previously (aws.operation, rpc.method), are now able to be annotations when explicitly listed.

Unit test cases:
- indexedAttrs includes special attributes, do not index all -> should annotate special attributes
- indexedAttrs includes special attributes, do index all -> should annotate special attributes
- indexedAttrs doesn't includes special attributes, do index all -> should NOT annotate special attributes

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/awsxrayexporter/internal/translator/segment.go [hunk 1]
```diff
@@ -320,6 +321,19 @@ func timestampToFloatSeconds(ts pcommon.Timestamp) float64 {
 	return float64(ts) / float64(time.Second)
 }
 
+func addSpecialAttributes(attributes map[string]pcommon.Value, indexedAttrs []string, unfilteredAttributes pcommon.Map) map[string]pcommon.Value {
+	for _, name := range indexedAttrs {
+		// Index attributes that have been filtered out before if explicitly added
+		_, filteredIn := attributes[name]
```
Comment (atshaw43) [inline exporter/awsxrayexporter/internal/translator/segment.go:327]: Nit:
I found variable names like filteredIn a little confusing on their purpose. Could we use something like currentValue, overrideValue, hasOverrideValue, ect.

Comment (jj22ee) [reply exporter/awsxrayexporter/internal/translator/segment.go]: addressed

**Review by atshaw43 — APPROVED** [review]

**Review by willarmiros — APPROVED** [review]
@open-telemetry/collector-contrib-approvers @TylerHelmuth  this ready to merge!

**Comment by fatsheep9146** [comment]
Please help add a changlog @jj22ee

File: exporter/awsxrayexporter/internal/translator/segment.go [hunk 2]
```diff
@@ -108,15 +110,14 @@ func MakeSegment(span ptrace.Span, resource pcommon.Resource, indexedAttrs []str
...
 		service                                            = makeService(resource)
 		sqlfiltered, sql                                   = makeSQL(span, awsfiltered)
-		user, annotations, metadata                        = makeXRayAttributes(sqlfiltered, resource, storeResource, indexedAttrs, indexAllAttrs)
+		additionalAttrs                                    = addSpecialAttributes(sqlfiltered, indexedAttrs, attributes)
+		user, annotations, metadata                        = makeXRayAttributes(additionalAttrs, resource, storeResource, indexedAttrs, indexAllAttrs)
 		name                                               string
 		namespace                                          string
 	)
 
 	// X-Ray segment names are service names, unlike span names which are methods. Try to find a service name.
```
Comment (TylerHelmuth) [inline exporter/awsxrayexporter/internal/translator/segment.go:119]: Where should this comment be now?

Comment (jj22ee) [reply exporter/awsxrayexporter/internal/translator/segment.go:119]: This comment is for the next 80~ lines, where the value of `name` is attempted to be found.

**Review by TylerHelmuth — APPROVED** [review]
Needs a changelog entry still

**Pull request merged**